### PR TITLE
Add curry of N arguments

### PR DESCRIPTION
**Pull request by grvcoelho**
@MarcoWorms, @joaolucasl :tada:

### CODE REVIEW

File: lib/curry.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { isPlaceholder } from './placeholder'
+
+const merge = (dest, origin) => {
+	const newArgs = dest.map(i => {
```
Comment (MarcoWorms) [inline lib/curry.js]: you could normalize one-parameter arrow funcions with either "(x)" or "x". "actualLength" is also a one-parameter arrow function and you did wrap it in (), here you aren't :)

File: lib/curry.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	})
+
+	return newArgs.concat(origin)
+}
+
+const actualLength = (arr) =>
+	arr.reduce((len, item) => isPlaceholder(item) ? len : len + 1, 0)
+
+const curry = (fn, arity = fn.length, received = []) => (...args) => {
+	const combined = merge(received.slice(), args.slice())
```
Comment (MarcoWorms) [inline lib/curry.js]: I know slice() is a native way of cloning arrays, but I would make this explicit like
```
const clone = array => array.slice()
```
It makes the code more self-explanatory, and it's also a cool new function for your lib :D (you can also implement id `x => x` and use `const clone array => array.map(id)` for 100% hipsterness)

File: lib/curry.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { isPlaceholder } from './placeholder'
+
+const merge = (dest, origin) => {
+	const newArgs = dest.map(i => {
+		let elem = i
```
Comment (MarcoWorms) [inline lib/curry.js]: `elem`ental mage?

File: lib/curry.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+			elem = origin.shift()
+		}
+
+		return elem
+	})
+
+	return newArgs.concat(origin)
+}
+
+const actualLength = (arr) =>
```
Comment (MarcoWorms) [inline lib/curry.js]: `arr` i'm a pirate?

File: lib/curry.js (hunk 3 above)
Comment (MarcoWorms) [reply lib/curry.js]: `i`like potato?

**Comment by grvcoelho** [comment]
@MarcoWorms HAHAHAHAHHA

Best PR review ever!

**Pull request merged**